homeassistant/components/zha/core/gateway.py

```diff
@@ -231,12 +233,17 @@ async def async_initialize(self) -> None:
 
     def connection_lost(self, exc: Exception) -> None:
         """Handle connection lost event."""
+        _LOGGER.debug("Connection to the radio was lost: %r", exc)
+
         if self.shutting_down:
             return
 
-        _LOGGER.debug("Connection to the radio was lost: %r", exc)
+        # Ensure we do not queue up multiple resets
+        if self._reload_task is not None and not self._reload_task.done():
```

Comment:
Why even check `done()`. If a reload completes, isn't this whole class garbage collected and build up again with the new config entry ?